### ISSUE TITLE
Cannot read property 'getAttribute' of null

### ISSUE THREAD

**Issue by fmay** [ckeditor/ckeditor5-vue#24]
Hi - I'm looking to replace Froala and am doing a POC with CKEditor. Unfortunately I'm getting the above error when running the test. It all builds fine.

I am using Vue with Nuxt.

I have followed [the guide](https://ckeditor.com/docs/ckeditor5/latest/builds/guides/integration/frameworks/vuejs.html#using-ckeditor-from-source) to the letter (I think).

Here's the error

```
ckeditor.js:5 TypeError: Cannot read property 'getAttribute' of null
    at IconView._updateXMLContent (iconview.js:100)
    at IconView.render (iconview.js:76)
    at IconView.on (observablemixin.js:254)
    at IconView.fire (emittermixin.js:196)
    at IconView.(anonymous function) [as render] (http://localhost:4000/_nuxt/vendors.pages/test.js:59093:16)
    at ViewCollection.on (viewcollection.js:68)
    at ViewCollection.fire (emittermixin.js:196)
    at ViewCollection.add (collection.js:182)
    at ButtonView.render (buttonview.js:168)
    at ButtonView.on (observablemixin.js:254)
```

Here's my vue file

```
<template>
  <div>
    <ckeditor :editor="editor" v-model="editorData" :config="editorConfig"></ckeditor>
  </div>
</template>

<script>
import CKEditor from '@ckeditor/ckeditor5-vue'
import BalloonEditor from '@ckeditor/ckeditor5-editor-balloon/src/ballooneditor'
import ClassicEditor from '@ckeditor/ckeditor5-editor-classic/src/classiceditor'
import EssentialsPlugin from '@ckeditor/ckeditor5-essentials/src/essentials'
import BoldPlugin from '@ckeditor/ckeditor5-basic-styles/src/bold'
import ItalicPlugin from '@ckeditor/ckeditor5-basic-styles/src/italic'
import LinkPlugin from '@ckeditor/ckeditor5-link/src/link'
import ParagraphPlugin from '@ckeditor/ckeditor5-paragraph/src/paragraph'

export default {
  data() {
    return {
      editor: ClassicEditor,
      editorData: '<p>Content of the editor.</p>',
      editorConfig: {
        plugins: [
          EssentialsPlugin,
          BoldPlugin,
          ItalicPlugin,
          LinkPlugin,
          ParagraphPlugin
        ],

        toolbar: {
          items: ['bold', 'italic', 'link', 'undo', 'redo']
        }
      }
    }
  }
}
</script>
```

and my `vue.config.js`

```
const CKEditorWebpackPlugin = require( '@ckeditor/ckeditor5-dev-webpack-plugin' );
const { styles } = require( '@ckeditor/ckeditor5-dev-utils' );

module.exports = {
  // The source of CKEditor is encapsulated in ES6 modules. By default, the code
  // from the node_modules directory is not transpiled, so you must explicitly tell
  // the CLI tools to transpile JavaScript files in all ckeditor5-* modules.
  transpileDependencies: [
      /ckeditor5-[^/\\]+[/\\]src[/\\].+\.js$/,
  ],

  configureWebpack: {
      plugins: [
          // CKEditor needs its own plugin to be built using webpack.
          new CKEditorWebpackPlugin( {
              // See https://ckeditor.com/docs/ckeditor5/latest/features/ui-language.html
              language: 'en'
          } )
      ],
      devtool: '#source-map'
  },

  css: {
      loaderOptions: {
          // Various modules in the CKEditor source code import .css files.
          // These files must be transpiled using PostCSS in order to load properly.
          postcss: styles.getPostCssConfig( {
              themeImporter: {
                  themePath: require.resolve( '@ckeditor/ckeditor5-theme-lark' )
              },
              minify: true
          } )
      }
  },

  chainWebpack: config => {
      // Vue CLI would normally use its own loader to load .svg files. The icons used by
      // CKEditor should be loaded using raw-loader instead.
      config.module
          .rule( 'svg' )
          .test( /ckeditor5-[^/\\]+[/\\]theme[/\\]icons[/\\][^/\\]+\.svg$/ )
          .use( 'file-loader' )
          .loader( 'raw-loader' );
  }
};
```

What have I done wrong!!!

Thanks

**Comment by oleq** [comment]
It looks like the SVG icons used by CKEditor were not loaded correctly by the `raw-loader`. We didn't test Nuxt and I can't tell what is wrong with it but I'd search and debug around `chainWebpack`. Can you create a minimal example that reproduces the issue and publish it, for instance, on GitHub so we can take a look on it?

**Comment by nqdy666** [comment]
I had the same problem  -.-

**Comment by fmay** [comment]
I actually carried on using Froala for the time being until this gets fixed. I am looking forward to using CKEditor but I won't spend the time hacking it into shape until this works out of the box.

@oleq if you can confirm that it works with Nuxt then I will make the switchover.

**Comment by oleq** [comment]
@fmay I'm not really into Nuxt but I managed to get rid of errors and get CSS compiled using this `nux.config.js` (no `vue.config.js`). The trick is excluding CKEditor CSS from default rules defined in Nuxt's Webpack config. I hope this will help you.

P.S. have your say in https://github.com/ckeditor/ckeditor5/issues/1511.

```js
const pkg = require('./package')
const CKEditorWebpackPlugin = require( '@ckeditor/ckeditor5-dev-webpack-plugin' );
const { styles } = require( '@ckeditor/ckeditor5-dev-utils' );

module.exports = {
  mode: 'universal',

  /*
  ** Headers of the page
  */
  head: {
    title: pkg.name,
    meta: [
      { charset: 'utf-8' },
      { name: 'viewport', content: 'width=device-width, initial-scale=1' },
      { hid: 'description', name: 'description', content: pkg.description }
    ],
    link: [
      { rel: 'icon', type: 'image/x-icon', href: '/favicon.ico' }
    ]
  },

  /*
  ** Customize the progress-bar color
  */
  loading: { color: '#fff' },

  /*
  ** Global CSS
  */
  css: [
  ],

  /*
  ** Plugins to load before mounting the App
  */
  plugins: [
  ],

  /*
  ** Nuxt.js modules
  */
  modules: [
  ],

  /*
  ** Build configuration
  */
  build: {
    /*
    ** You can extend webpack config here
    */
    extend(config, ctx) {
      config.module.rules.forEach( rule => {
        if ( isCssRule( rule ) ) {
          rule.exclude = /ckeditor5-[^/]+\/theme\/[\w-/]+\.css$/;
        }
      } );

      config.module.rules.unshift(
        {
          // Or /ckeditor5-[^/]+\/theme\/icons\/[^/]+\.svg$/ if you want to limit this loader
          // to CKEditor 5 icons only.
          test: /ckeditor5-[^/]+\/theme\/icons\/[^/]+\.svg$/,
          use: [ 'raw-loader' ]
        },
        {
          // Or /ckeditor5-[^/]+\/theme\/[\w-/]+\.css$/ if you want to limit this loader
          // to CKEditor 5 theme only.
          test: /ckeditor5-[^/]+\/theme\/[\w-/]+\.css$/,
          use: [
            {
              loader: 'style-loader',
              options: {
                singleton: true
              }
            },
            {
              loader: 'postcss-loader',
              options: styles.getPostCssConfig( {
                themeImporter: {
                  themePath: require.resolve( '@ckeditor/ckeditor5-theme-lark' )
                },
                minify: true
              } )
            },
          ]
        }
      );

        // CKEditor needs its own plugin to be built using webpack.
      config.plugins.unshift(
          new CKEditorWebpackPlugin( {
              // See https://ckeditor.com/docs/ckeditor5/latest/features/ui-language.html
              language: 'en'
          } )
      );
    }
  }
}

function isCssRule( rule ) {
  return rule.test.toString().indexOf( 'css' ) > -1;
}
```

**Comment by fmay** [comment]
That's great - thanks for doing this. I will look to try it out and integrate it asap.

Many thanks!

**Comment by kishorpatel85** [comment]
getting the same type of error in vue

TypeError: Cannot read property 'getAttribute' of null
    at IconView._updateXMLContent (app.js:49634)
    at IconView.render (app.js:49610)
    at IconView.<anonymous> (app.js:61248)
    at IconView.fire (app.js:59117)
    at IconView.<computed> [as render] (app.js:61252)
    at ViewCollection.<anonymous> (app.js:53899)
    at ViewCollection.fire (app.js:59117)
    at ViewCollection.add (app.js:55546)
    at ButtonView.render (app.js:47331)
    at ButtonView.<anonymous> (app.js:61248)

**Comment by nitish1986** [comment]
@kishorpatel85 Did you get any solution to your problem? I'm also facing the same issue, this comes only when I try to include `@wiris/mathtype-ckeditor5` or else it works perfectly fine on my local builds.

**Comment by huangluloveTing** [comment]
next? @nitish1986

**Comment by mohitsehgal** [comment]
> @kishorpatel85 Did you get any solution to your problem? I'm also facing the same issue, this comes only when I try to include `@wiris/mathtype-ckeditor5` or else it works perfectly fine on my local builds.

Same Problem for me. @Author please do something

**Comment by pinktoadette** [comment]
same problem in angular

**Comment by mi-lopez** [comment]
> > @kishorpatel85 Did you get any solution to your problem? I'm also facing the same issue, this comes only when I try to include `@wiris/mathtype-ckeditor5` or else it works perfectly fine on my local builds.
> 
> Same Problem for me. @author please do something

Same problem in VueJS 2+

**Comment by crying2812** [comment]
I have same issues, tried many diffrent ways but it wont work :(

**Comment by felippi** [comment]
I have the same problem with Vue 2

**Comment by FilipTokarski** [comment]
For anyone experiencing problems with Vue and Wiris MathType plugin, please see [this solution](https://github.com/ckeditor/ckeditor5/issues/2048#issuecomment-533896390), it might help.

**Comment by JarvisH** [comment]
I had the same issue during vue-cli tests, the problem turned out to be `moduleNameMapper` in the jest config file, where all svgs where being mocked/stubbed.

**Comment by sadeq-qafari** [comment]
After so many attempts this helped me fix the problem:
https://github.com/ckeditor/ckeditor5/issues/6071#issuecomment-866209780
just needed the webpack rule like this:
```
        const filesRuleIndex = cfg.module.rules.findIndex(item => {
          return item.test.test('.svg')
        })

        if (filesRuleIndex !== -1) {
          cfg.module.rules[filesRuleIndex].test = /\.(png|jpe?g|gif|webp)$/
          const svgRule = {...cfg.module.rules[filesRuleIndex]}
          svgRule.test = /\.svg/
          svgRule.exclude = svgRule.exclude || []
          svgRule.exclude.push(path.join(__dirname, 'node_modules', '@ckeditor'))
          cfg.module.rules.push(svgRule)
        }

        cfg.module.rules.push({
          test: /ckeditor5-[^/\\]+[/\\]theme[/\\]icons[/\\][^/\\]+\.svg$/,
          use: ["raw-loader"]
        })
```

**Comment by AlejandroBryan** [comment]
> After so many attempts this helped me fix the problem: [ckeditor/ckeditor5#6071 (comment)](https://github.com/ckeditor/ckeditor5/issues/6071#issuecomment-866209780) just needed the webpack rule like this:
> 
> ```
>         const filesRuleIndex = cfg.module.rules.findIndex(item => {
>           return item.test.test('.svg')
>         })
> 
>         if (filesRuleIndex !== -1) {
>           cfg.module.rules[filesRuleIndex].test = /\.(png|jpe?g|gif|webp)$/
>           const svgRule = {...cfg.module.rules[filesRuleIndex]}
>           svgRule.test = /\.svg/
>           svgRule.exclude = svgRule.exclude || []
>           svgRule.exclude.push(path.join(__dirname, 'node_modules', '@ckeditor'))
>           cfg.module.rules.push(svgRule)
>         }
> 
>         cfg.module.rules.push({
>           test: /ckeditor5-[^/\\]+[/\\]theme[/\\]icons[/\\][^/\\]+\.svg$/,
>           use: ["raw-loader"]
>         })
> ```

Hi,
may you be able to explane ?
how it's works for with plane js and webpack without a core ?
i am having the same issue event thought  my bundle compile without errors.

**Comment by alext100** [comment]
Had this problem when installed and then uninstalled some plugins. Some packages was deleted with them. Helped just `npm i`
